propagation/propagation.go

```diff
@@ -40,6 +40,29 @@ type TextMapCarrier interface {
 	// must never be done outside of a new major release.
 }
 
+// BytesMapCarrier is a Carrier that stores propagated
```

Comment:
I'm missing something -- why use this class and not `HeaderCarrier`? Would you extend the comment to help explain? I imagine there's some MIME encoding restrictions if you use HeaderCarrier that you are trying to avoid -- if so, what are you actually using this for if I may ask?